frontend/src/container/WorkspaceContent.jsx

```diff
@@ -598,9 +600,48 @@ export class WorkspaceContent extends React.Component {
     props.history.push(props.location.pathname + '?' + qs.stringify(newUrlSearchObject, { encode: false }))
   }
 
-  filterWorkspaceContent = (contentList, filter) => filter.length === 0
-    ? contentList
-    : contentList.filter(c => c.type === CONTENT_TYPE.FOLDER || filter.includes(c.type)) // keep unfiltered files and folders
+  filterWorkspaceContentByUserInput = (contentList, userFilter) => {
+    const { props } = this
+
+    const matchesUserInput = (content) => {
+      const contentTypeInfo = props.contentType.find(info => info.slug === content.type)
+      const statusInfo = contentTypeInfo.availableStatuses.find(
+        s => s.slug === content.statusSlug
+      )
+
+      return content.label.toUpperCase().includes(userFilter.toUpperCase()) ||
+        content.lastModifier.public_name.toUpperCase().includes(userFilter.toUpperCase()) ||
+        props.t(contentTypeInfo.label).toUpperCase().includes(userFilter.toUpperCase()) ||
+        props.t(statusInfo.label).toUpperCase().includes(userFilter.toUpperCase())
+    }
+
+    const userFilteredList = userFilter === ''
+      ? contentList
+      : contentList.filter(c =>
+        matchesUserInput(c) ||
+        c.type === CONTENT_TYPE.FOLDER
+      )
+
+    const folderSet = new Set()
+    userFilteredList.map(c => { if (c.parentId !== null) folderSet.add(c.parentId) })
+
+    return userFilter === ''
+      ? userFilteredList
+      : userFilteredList.filter(c =>
+        c.type !== CONTENT_TYPE.FOLDER ||
+          (c.type === CONTENT_TYPE.FOLDER &&
+          (folderSet.has(c.id) || matchesUserInput(c)))
+      )
+  }
+
+  filterWorkspaceContent = (contentList, filter, userFilter) => {
+    const userFilteredList = this.filterWorkspaceContentByUserInput(contentList, userFilter)
+
+    return filter.length === 0
+      ? userFilteredList
+      : userFilteredList.filter(c => c.type === CONTENT_TYPE.FOLDER ||
+        filter.includes(c.type))
+  } // keep unfiltered files and folders
```

Comment:
Forgotten comment

Reply:
This one was already here before, so I don't know if I should delete it or keep it

Reply:
It seems to me to be more someone's forgetfulness than documentation. When you went to modify the function, was this comment useful? If yes, keep it, if no, you can delete it.